### ISSUE TITLE
Unclear deprection suggestions for run_hash

### ISSUE THREAD

**Issue by vakker** [aimhubio/aim#2190]
## 🐛 Bug

When I set the `run_hash` manually I get:
```
Setting custom `Run.hash` value is deprecated! Consider setting `Run.name` instead. This functionality will be removed in version 3.15.
```

### To reproduce
```python
aim.Run(experiment="asd", run_hash="potato")
```

### Expected behavior

The warning is clear per se, however the fix is not. 
It comes from [here](https://github.com/aimhubio/aim/blob/ac930b63641c061e06ea82992def7afaa8553639/aim/sdk/base_run.py#L42).
The `Run` class doesn't have a `name` property, nor a parameter to its `__init__`.
Am I missing something?

### Environment

- Aim Version: v3.13.3
- Python version: 3.9.7

**Comment by alberttorosyan** [comment]
Hey @vakker! Adding the name argument to `Run.__init__` will resolve the issue. Meanwhile, can you try to set the `name` right after creating `Run` instance?
```python
run = aim.Run(experiment="asd")
run.name = "potato"
```

**Comment by vakker** [comment]
Yeah, setting it after the init works. But then is it used anywhere? The `Run` class doesn't seem to use the `name` property anywhere.

**Comment by alberttorosyan** [comment]
@vakker the `Run.name` is displayed in UI (see attached screenshot), as well as can be used to filter runs, metrics when doing queries:
`run.name == "my run" and metric.context.subset == "train"`

<img width="1707" alt="Screen Shot 2022-09-22 at 14 25 19" src="https://user-images.githubusercontent.com/32957250/191723354-eecb60b7-e31c-4c84-946a-dc421e5ae836.png">

**Comment by alberttorosyan** [comment]
as for the setting the name via `__init__`, could you please change the issue description accordingly? alternative would be to submit the new feature request and close this one.

**Comment by alberttorosyan** [comment]
Hey @vakker! Sorry for late response; somehow missed your last comment. I think something like this would be a good match:
`Add support for setting Run name via Run.__init__`

**Comment by sharathmk9** [comment]
Hi @SGevorg, We have developed this features internally that user can choose the optional run name `Run(name='my-run')` or by setting env variable `export __AIM_RUN_NAME__=my-run`
If no one has started the development, we can plan to move the changes to this repo. Thanks

**Comment by Pyrestone** [comment]
FYI, `run.description` is a parameter that works fine for me and is displayed in the runs explorer.

e.g. I've been using my config name and job ID for this purpose:

```python
aim_run=Run(repo=repo_path, experiment=experiment_name)
aim_run.description = "config: " + config_name + " ( Job ID " + slurm_job_id + ")"
```

no changes to aim code needed for this workaround.
